pkgs/by-name/ku/kubernetes-kcp/package.nix

```diff
@@ -0,0 +1,68 @@
+{
+  lib,
+  stdenv,
+  buildGoModule,
+  fetchFromGitHub,
+  installShellFiles,
+  testers,
+}:
+
+buildGoModule rec {
+  pname = "kubernetes-kcp";
+  version = "0.26.0";
+
+  src = fetchFromGitHub {
+    owner = "kcp-dev";
+    repo = "kcp";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-ZEgDeILo2weSAZgBsfR2EQyzym/I/+3P99b47E5Tfrw=";
+  };
+  vendorHash = "sha256-IONbTih48LKAiEPFNFdBkJDMI2sjHWxiqVbEJCskyio=";
+
+  subPackages = [ "cmd/kcp" ];
+
+  # TODO: The upstream has the additional version information pulled from go.mod
+  # dependencies.
+  ldflags = [
+    "-X k8s.io/client-go/pkg/version.gitCommit=unknown"
+    "-X k8s.io/client-go/pkg/version.gitTreeState=clean"
+    "-X k8s.io/client-go/pkg/version.gitVersion=v${version}"
+    # "-X k8s.io/client-go/pkg/version.gitMajor=${KUBE_MAJOR_VERSION}"
+    # "-X k8s.io/client-go/pkg/version.gitMinor=${KUBE_MINOR_VERSION}"
```

Comment:
You could use `lib.versions.major kubernetes.version` probably (same goes for minor)

Reply:
Ah thanks for the note, I'll check more into that! It is probably going to be a bit more involved, though, as this needs to pick up the version details from `go.mod`. That probably has to be done preInstall phase, or even by adjusting the installPhase. I do not think this is a must have, and we could look to improve and match the upstream deployment setup later down the line